storage/store.go

```diff
@@ -2506,6 +2511,8 @@ func (s *Store) processRaft() {
 			s.pendingRaftGroups.Unlock()
 			s.mu.Unlock()
 
+			s.processRaftMu.Lock()
```

Comment:
We almost have that guarantee. `Replica.withRaftGroupLocked` returns immediately if `Replica.mu.destroyed` is non-nil. The only missing piece is that we don't always call `Replica.Destroy` when removing a replica, but that is easy to change.